File: ios/NativeComponent/BrightcovePlayer/VolumeView/CustomVolumeView.m
//
//  CustomVolumeView.m
//  OnAppTV
//
//  Created by Chuong Huynh on 3/2/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "CustomVolumeView.h"
#import "UIView+Extension.h"

@interface CustomVolumeView()

@property (strong, nonatomic) IBOutlet UIView *contentView;

@end

@implementation CustomVolumeView

- (instancetype)init {
  self = [super init];
  return self;
}

- (instancetype)initWithFrame:(CGRect)frame {
  
  self = [super initWithFrame:frame];
  [self commonInit];
  return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
  self = [super initWithCoder:aDecoder];
  [self commonInit];
  return self;
}

- (void)commonInit {
  
  [[NSBundle mainBundle] loadNibNamed:@"CustomVolumeView" owner:self options:nil];
  [self addSubview:_contentView];
  _contentView.frame = self.bounds;
  _contentView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
}

@end

File: ios/NativeComponent/BrightcovePlayer/VolumeView/CustomVolumeView.swift
//
//  CustomVolumeView.swift
//  OnAppTV
//
//  Created by Chuong Huynh on 3/4/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

import Foundation
import UIKit

class CustomVolumeView: UIView {
    
    // MARK: - Outlets
    @IBOutlet weak var volumeSlider: SliderMPVolumeView!
    @IBOutlet weak var quieterIcon: UIImageView!
    @IBOutlet weak var louderIcon: UIImageView!
    @IBOutlet var contentView: UIView!
    
    // MARK: - Constructors
    override init(frame: CGRect) {
        super.init(frame: frame)
        commonInit()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        commonInit()
    }
    
    convenience init() {
        self.init(frame: .zero)
    }
    
    func commonInit() {
        Bundle.main.loadNibNamed("CustomVolumeView", owner: self, options: nil)
        self.addSubview(contentView)
        contentView.frame = self.bounds
        contentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
    }
}
